src/rgw/rgw_sal_rados.h

```diff
@@ -383,6 +383,10 @@ class RadosObject : public StoreObject {
 
     virtual ~RadosObject();
 
+    virtual void invalidate() override {
+      StoreObject::invalidate();
+      rados_ctx->invalidate(get_obj());
```

Comment:
i see that `RadosObject` has a default ctor that leaves this `rados_ctx` uninitialized. should we initialize it to `nullptr` there and add a null check here? or maybe drop the default ctor?

Reply:
Good question.  I'll look at dropping it.